File: app/src/main/java/com/example/olesya/boardgames/connection/BoundService.java
package com.example.olesya.boardgames.connection;

import android.app.Service;
import android.arch.lifecycle.MutableLiveData;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class BoundService extends Service {

    protected final int PORT_NUMBER = 8888;

    protected MutableLiveData<String> serviceMessage = new MutableLiveData<>();

    protected MyBinder binder = new BoundService.MyBinder();

    public void setMessage(MutableLiveData<String> message) {
        this.serviceMessage = message;
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    public class MyBinder extends Binder {
        BoundService getService() {
            return BoundService.this;
        }
    }
}

File: app/src/main/java/com/example/olesya/boardgames/connection/BoundService.kt
package com.example.olesya.boardgames.connection

import android.app.Service
import android.arch.lifecycle.MutableLiveData
import android.content.Intent
import android.os.Binder
import android.os.IBinder

open class BoundService: Service() {

    val PORT_NUMBER = 8888

    var serviceMessage: MutableLiveData<String> = MutableLiveData()

    val binder: BoundService.MyBinder = MyBinder()

    override fun onBind(intent: Intent?): IBinder? {
        return binder
    }

    inner class MyBinder: Binder() {
        fun getService(): BoundService {
            return this@BoundService
        }
    }

}